src/index.js

```diff
@@ -367,7 +367,9 @@ module.exports = function flowReactPropTypes(babel) {
         return;
       }
 
-      valueNode = propTypesAST;
+      // Clone the props node so that we don't use the same node in multiple
+      // parts of the same AST.
+      valueNode = JSON.parse(JSON.stringify(propTypesAST));
```

Comment:
I asked in babel slack and there's `babel.types.cloneDeep(propTypesAST)` for this use case.